web/client/utils/FeatureGridUtils.js

```diff
@@ -162,19 +162,50 @@ module.exports = {
     applyAllChanges: (orig, changes = {}) => applyChanges(orig, changes[orig.id] || {}),
     applyChanges,
     gridUpdateToQueryUpdate: ({attribute, operator, value, type} = {}, oldFilterObj = {}) => {
+        const andGroup = {
+            id: 9999, // dummy number representing group id
+            logic: "AND",
+            index: 0
+        };
+
+        let mergeFilters = oldFilterObj.groupFields 
+            ? (oldFilterObj.groupFields.filter(obj => obj.id === 1 ).length > 0 && oldFilterObj.filterFields.length > 0 
+               ? true // checks if query panel filter is active
+                : oldFilterObj.filterFields.length - 1 > 0 
+                    || ( oldFilterObj.filterFields[0] && oldFilterObj.filterFields[0].hasOwnProperty('loading'))) 
```

Comment:
why you check loading?

Reply:
All filter objects comming from query panel filter have loading property, but those filter objects coming from Featuregrid filter don't have loading property. therefore, I have used loading property to differentiate between filter objects comming from query panel filter and those filter objects coming from Featuregrid filter

Reply:
So the oldFilter contains also the filters added by the inline filter tool?
If so, please mark the filterFields coming from the inline filter tool (or only the group) accordingly when you create the new filter. Using the loading property is an hack and it is hard to maintain and to understand. 
Is there any other possibility to keep separated the 2 entities ?

Reply:
sorry, what do you mean by "inline filter tool", I have not understood.

Reply:
I have not understood what do you mean by "filterFields coming from the inline filter tool". I would like to use your adivice. please help me to understand this "inline filter tool".